monkey/common/configuration/__init__.py

```diff
@@ -1,6 +1,15 @@
-from .agent_configuration import (
-    AgentConfiguration,
-    AgentConfigurationSchema,
+from .agent_configuration import AgentConfiguration, InvalidConfigurationError
+from .agent_sub_configurations import (
+    CustomPBAConfiguration,
+    PluginConfiguration,
+    ScanTargetConfiguration,
+    ICMPScanConfiguration,
+    TCPScanConfiguration,
+    NetworkScanConfiguration,
+    ExploitationOptionsConfiguration,
+    ExploiterConfiguration,
+    ExploitationConfiguration,
+    PropagationConfiguration,
```

Comment:
Why do we have both `PluginConfiguration` and `ExploiterConfiguration` dataclasses which have the same member variables ([link to file with definitions](https://github.com/guardicore/monkey/blob/1960-deserialize-config/monkey/common/configuration/agent_sub_configurations.py))?

Reply:
Good question! `ExploiterConfiguration` had `supported_os` in it, until that was removed by #2038. When `supported_os` was removed, everything that used `ExploiterConfiguration` should have been changed to `PluginConfiguration`. I'll add a task to #1960.